app-creator/client/src/redux/reducer.ts

```diff
@@ -203,8 +226,14 @@ export const reducer: Reducer<AppCreatorStore, AppCreatorAction | AnyAction> = (
       };
     case SET_SELECTED_TEMPLATE:
       addDefaultStyles(action.payload.template);
+      applyPalette(action.payload.template.widgets, state.selectedPalette.name);
       return {
         ...INITIAL_STATE,
+        widgetIDs:
+          state.eventType === EventType.changingPalette
+            ? state.widgetIDs
+            : INITIAL_STATE.widgetIDs,
```

Comment:
What's the difference between using either widgethIDs?

Reply:
`SET_SELECTED_TEMPLATE` is called to reset the template with all its state. So when we select new templates from the templates tab, or when we import a new template. In addition it is also called when changing palettes. However, in the case of palette change, we do not want to reset the widgetIDs.